packages/eas-cli/src/metadata/apple/config/writer.ts

```diff
@@ -64,15 +66,53 @@ export class AppleConfigWriter {
     };
   }
 
-  public setCategories({ primaryCategory, secondaryCategory }: AttributesOf<AppInfo>): void {
-    this.schema.categories = [];
+  public setCategories(
+    attributes: Pick<
+      AttributesOf<AppInfo>,
+      | 'primaryCategory'
+      | 'primarySubcategoryOne'
+      | 'primarySubcategoryTwo'
+      | 'secondaryCategory'
+      | 'secondarySubcategoryOne'
+      | 'secondarySubcategoryTwo'
+    >
+  ): void {
+    this.schema.categories = undefined;
+    if (!attributes.primaryCategory) {
+      return;
+    }
 
-    // TODO: see why these types are conflicting
-    if (primaryCategory) {
-      this.schema.categories.push(primaryCategory.id as any);
-      if (secondaryCategory) {
-        this.schema.categories.push(secondaryCategory.id as any);
+    // We are receiving strings, but need AppCategoryId or AppSubcategoryId.
+    // These are returned by the ASC API and should be valid categories.
+    const primaryCat = attributes.primaryCategory.id as AppCategoryId;
+    const primarySubOne = attributes.primarySubcategoryOne?.id as AppSubcategoryId | undefined;
+    const primarySubTwo = attributes.primarySubcategoryTwo?.id as AppSubcategoryId | undefined;
+    const secondaryCat = attributes.secondaryCategory?.id as AppCategoryId | undefined;
+    const secondarySubOne = attributes.secondarySubcategoryOne?.id as AppSubcategoryId | undefined;
+    const secondarySubTwo = attributes.secondarySubcategoryTwo?.id as AppSubcategoryId | undefined;
```

Comment:
I kinda hate doing this, but the types are strict. Our model has `string` types for each attribute, but our schema types have `AppCategoryId | AppSubcategoryId`.

Alternatively, we could also relax the schema type to just match `string`. This is validated anyways in the schema :) cc @EvanBacon

Reply:
nvm, I'm going to relax the types.